src/main/java/seedu/modtrek/logic/parser/DeleteCommandParser.java

```diff
@@ -17,9 +22,22 @@ public class DeleteCommandParser implements Parser<DeleteCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public DeleteCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argumentMultimap = ArgumentTokenizer.tokenize(args, PREFIX_CODE);
         try {
-            Code code = ParserUtil.parseCode(args);
-            return new DeleteCommand(code);
+            boolean isAll;
+            Set<Code> codes;
+            String flag = argumentMultimap.getPreamble();
+            if (flag.equals("all")) {
```

Comment:
I think there needs to add a check for any arguments after delete all.

```suggestion
            if (flag.equals("all") && argumentMultimap.getValue(PREFIX_CODE).isEmpty()) {
```

Reply:
oops nvm ignore this comment.